src/NumericalAlgorithms/Spectral/ComplexDataView.cpp

```diff
@@ -5,56 +5,83 @@
 
 #include "DataStructures/ComplexDataVector.hpp"
 #include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Assert.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/Math.hpp"  // IWYU pragma: keep
 
 namespace Spectral {
 namespace SWSH {
 namespace detail {
 
+template <typename T>
+void sizes_assert(const T& vector, const size_t size) {
+  ASSERT(vector.size() == size,
+         "Assignment must be to the same size,"
+         " not "
+             << vector.size() << " assigned to " << size);
+}
+
+template void sizes_assert<DataVector>(const DataVector&, const size_t);
```

Comment:
You don't need the explicit instantiations. The compiler will instantiate the template whenever it has the full definition, which it will everywhere in this file :)